Source/Orts.Simulation/Simulation/RollingStocks/MSTSLocomotive.cs

```diff
@@ -4851,7 +5000,240 @@ public virtual float GetDataOf(CabViewControl cvc)
                 case CABViewControlTypes.ORTS_TCS48:
                     data = TrainControlSystem.CabDisplayControls[(int)cvc.ControlType - (int)CABViewControlTypes.ORTS_TCS1];
                     break;
-
+                case CABViewControlTypes.ORTS_SELECTED_SPEED:
+                case CABViewControlTypes.ORTS_SELECTED_SPEED_DISPLAY:
+                    {
+                        if (CruiseControl == null)
+                        {
+                            data = 0;
+                            break;
+                        }
+                        bool metric = cvc.Units == CABViewControlUnits.KM_PER_HOUR;
+                        float temp = CruiseControl.RestrictedSpeedActive ? MpS.FromMpS(CruiseControl.CurrentSelectedSpeedMpS, metric) : temp = MpS.FromMpS(CruiseControl.SelectedSpeedMpS, metric);
```

Comment:
```suggestion
                        float temp = CruiseControl.RestrictedSpeedActive ? MpS.FromMpS(CruiseControl.CurrentSelectedSpeedMpS, metric) : MpS.FromMpS(CruiseControl.SelectedSpeedMpS, metric);
                        
```
Just a little style change

Reply:
I was very tired to write this :D